ts/defs.bzl

```diff
@@ -172,6 +173,12 @@ def ts_project(
 
             See [docs/transpiler.md](/docs/transpiler.md) for more details.
 
+        declaration_transpiler: A custom transpiler tool to run that produces the TypeScript declaration outputs instead of `tsc`.
```

Comment:
nit: clarify that there are cases where `declaration_transpiler` is set but not used if the rule is configured to not output declarations, such as when `declaration` is False or when `no_emit` is True.

Reply:
may as well clarify the same thing for `transpiler` that it is not used when `no_emit` is True or `emit_declarations_only` is True